### PR TITLE
Fix issues when slicing a triangle geometry

### PR DESCRIPTION
**Pull request by imikejackson**

### CODE REVIEW

File: DREAM3DReviewFilters/SliceTriangleGeometry.h [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+ * (including typical open source software licenses).
+ *
+ * This source code file was originally written by United States DoD employees. The
+ * original source code files are released into the Public Domain.
+ *
+ * Subsequent changes to the codes by others may elect to add a copyright and license
+ * for those changes.
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ */
 #ifndef _SliceTriangleGeometry_h_
 #define _SliceTriangleGeometry_h_
```
Comment (JDuffeyBQ) [inline DREAM3DReviewFilters/SliceTriangleGeometry.h]:
```suggestion
#pragma once
```

File: DREAM3DReviewFilters/SliceTriangleGeometry.cpp [hunk 2]
```diff
@@ -359,11 +366,12 @@ void SliceTriangleGeometry::determineBoundsAndNumSlices(float& minDim, float& ma
       maxDim = m_Zend;
     }
   }
-
-  m_NumberOfSlices = static_cast<MeshIndexType>((maxDim - minDim) / m_SliceResolution) + 1;
+  // TODO: Is this still correct? Why have the subtraction at all?
+  m_NumberOfSlices = static_cast<MeshIndexType>((maxDim - 0.0) / m_SliceResolution) + 1;
```
Comment (sdonegan) [inline DREAM3DReviewFilters/SliceTriangleGeometry.cpp:370]: I think this is right. but I'm a little confused looking over the code.  Do we have an example of an STL file where it was crashing?  Would be good for a unit test.

**Review by sdonegan — COMMENTED** [review]
One comment in line, also can we change the human label to "Slice Triangle Geometry" for consistency with the class name?

File: DREAM3DReviewFilters/SliceTriangleGeometry.cpp (hunk 2 above)
Comment (imikejackson) [reply DREAM3DReviewFilters/SliceTriangleGeometry.cpp:370]: I think we could certainly create a small STL file that is lifted off the Z=0 plane as a test.

**Pull request merged**